### ISSUE TITLE
Predict gives different result than call

### ISSUE THREAD

**Issue by SergioG-M** [keras-team/keras#15139]
Hi,
I have trained a model without an issue, and I am trying to load it and evaluate some samples. I have done it before with other models using the call method without an issue, but now I am just getting nans. However the predict method works fine. Any idea that what may cause this?

![image](https://user-images.githubusercontent.com/61322372/128679310-4e668654-acd0-48f2-bdfa-46dc49123221.png)

I'm using tensorflow 2.3.1

**Comment by jvishnuvardhan** [comment]
@SergioG-M It is interesting. Can you please share a simple standalone code to reproduce the issue? Thanks!

**Comment by bitnick10** [comment]
I use predict API years.Without train,just call predict and give me accuracy more than 80%.So I ssume predict did the backpropagation.

**Comment by SergioG-M** [comment]
> @SergioG-M It is interesting. Can you please share a simple standalone code to reproduce the issue? Thanks!

That will be hard, I am using EfficientNet with some custom modifications, and a private dataset with a lot of preprocessing. Let me see if I can reproduce it with the usual EfficientNet and some normal image and I will get back to you, it may take me few days though.

**Comment by qlzh727** [comment]
It seems that the input to predict and __call__ are different (one extra dim for np.newaxis involved), not sure whether that's the cause.

Normally predict and __call__ should produce the same result. Both predict and __call__ does NOT do back propagation, and they don't update weights.
